File: src/main/java/com/github/shuaiouke/redlibcommandfile/language/psi/RedLibCommandFile.java
package com.github.shuaiouke.redlibcommandfile.language.psi;

import com.github.shuaiouke.redlibcommandfile.language.RedLibCommandFileType;
import com.github.shuaiouke.redlibcommandfile.language.RedLibCommandLanguage;
import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.lang.Language;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;

import static com.github.shuaiouke.redlibcommandfile.language.RedLibCommandLanguage.*;

public class RedLibCommandFile extends PsiFileBase {
    public RedLibCommandFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, INSTANCE);
    }

    @Override
    public @NotNull FileType getFileType() {
        return RedLibCommandFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "RedLib Command File";
    }
}

File: src/main/kotlin/com/github/shuaiouke/redlibcommandfile/language/psi/RedLibCommandFile.kt
package com.github.shuaiouke.redlibcommandfile.language.psi

import com.github.shuaiouke.redlibcommandfile.language.RedLibCommandFileType
import com.github.shuaiouke.redlibcommandfile.language.RedLibCommandLanguage
import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider

class RedLibCommandFile(viewProvider: FileViewProvider): PsiFileBase(viewProvider, RedLibCommandLanguage) {
    override fun getFileType(): FileType {
        return RedLibCommandFileType
    }

    override fun toString(): String {
        return "RedLib Command File"
    }
}
